src/streaming/streamingclient.py

```diff
@@ -0,0 +1,178 @@
+        """
+        self.client.abort()
+
+    def send_data_as_gen(self, generator):
+        """Function to send binary audio data from a generator
+
+        :param generator: generator object that yields binary audio data
+        """
+        if not generator:
+            raise ValueError('generator must be provided')
+
+        if hasattr(self, 'request_thread'):
+            if self.request_thread.isAlive():
+                raise ValueError("""Data is still being sent and will interfere 
+                    with the responses.""")
+
+        self.request_thread = self.thread(self.RequestSending, [generator])
+        self.request_thread.start()
+
+    def send_data_as_filepath(self, filepath):
+        """Function to send audio data from a given file
+
+        :param filepath: string containing file location
+        """
+        if not filepath:
+            raise ValueError('filepath must be provided')
+
+        if hasattr(self, 'request_thread'):
+            if self.request_thread.isAlive():
+                raise ValueError("""Data is still being sent and will interfere 
+                    with the responses.""")
+
+        self.send_data_as_gen(self.make_gen(filepath))
+
+    def make_gen(self, filepath):
+        """Get generator of audio data from a given filepath
+        
+        :param filepath: path to file including file name as string
+        :returns: generator of audio bytes
+        """
+        if not filepath:
+            raise ValueError('filepath must be provided')
+
+        with io.open(filepath, 'rb') as stream:
+            while True:
+                piece = stream.read(8192)
+                if not piece:
+                    break
+                yield piece
+
+    def RequestSending(self, generator):
+        """Function used in a thread to send requests to the server.
+
+        :param generator: generator object yielding audio data
+        """
+        if not generator:
+            raise ValueError('generator must be provided')
+
+        for chunk in generator:
+            self.client.send_binary(chunk)
+
+        self.client.send("EOS")
+
+    def responses(self):
+        """A generator of reponses from the server. Yields the data decoded.
+        """
+        while True:
+            with self.client.readlock:
+                opcode, response = self.client.recv_data()
+            if six.PY3 and opcode == websocket.ABNF.OPCODE_TEXT:
+                yield response.decode("utf-8")
+            elif opcode == websocket.ABNF.OPCODE_TEXT:
+                yield response
+            elif opcode == websocket.ABNF.OPCODE_CLOSE:
+                if response and len(response) >= 2:
+                    code = 256 * six.byte2int(response[0:1]) + six.byte2int(response[1:2])
+                    reason = response[2:].decode('utf-8')
+                    self.on_close(code, reason)
+                return
+            else:
+                yield ''
+
+class SpeechThread(threading.Thread):
```

Comment:
It seems like you're trying to make this class generic when in reality when only use this for one purpose. I would simplify this by defining it in the class as a separate thread with only the responsibility of sending data from a generator.